src/java/org/apache/cassandra/config/registry/ConfigurationRegistry.java

```diff
@@ -0,0 +1,371 @@
+        names.put("transparent_data_encryption_options_enabled", "transparent_data_encryption_options.enabled");
+        names.put("transparent_data_encryption_options_iv_length", "transparent_data_encryption_options.iv_length");
+
+        return names;
+    }
+
+    private static class TransformSet<T, R> extends AbstractSet<R>
+    {
+        private final Set<T> set;
+        private final Function<T, R> transform;
+
+        public TransformSet(Set<T> set, Function<T, R> transform)
+        {
+            this.set = set;
+            this.transform = transform;
+        }
+
+        /** {@inheritDoc} */
+        @Override
+        public Iterator<R> iterator()
+        {
+            Iterator<T> iter = set.iterator();
+            return new AbstractIterator<R>()
+            {
+                @Override
+                protected R computeNext()
+                {
+                    if (!iter.hasNext())
+                        return endOfData();
+
+                    T entry = iter.next();
+                    return transform.apply(entry);
+                }
+            };
+        }
+        /** {@inheritDoc} */
+        @Override
+        public int size()
+        {
+            return set.size();
+        }
+    }
+
+    private static class PropertyChangeListenerWrapper<T> implements PropertyChangeListener<T>
+    {
+        private final PropertyChangeListener<T> listener;
+        private final Class<T> type;
+
+        public PropertyChangeListenerWrapper(PropertyChangeListener<T> listener, Class<T> type)
+        {
+            this.listener = listener;
+            this.type = type;
+        }
+
+        /** {@inheritDoc} */
+        @Override
+        public void onBeforeChange(String name, Object oldValue, Object newValue)
+        {
+            T castedOldValue = type.cast(oldValue);
+            T castedNewValue = type.cast(newValue);
+            listener.onBeforeChange(name, castedOldValue, castedNewValue);
+        }
+    }
+
+    /**
+     * Configuration enricher is used to add new property setters to the configuration.
+     */
+    private static class ConfigurationSettersEnricher implements PropertySetterWalker.SetterVisitor
+    {
+        private final Map<String, ConfigurationUnit<?>> registry;
+
+        ConfigurationSettersEnricher(Map<String, ConfigurationUnit<?>> registry)
+        {
+            this.registry = registry;
+        }
+
+        /** {@inheritDoc} */
+        @Override
+        public <T> void visit(String name, Class<T> clazz, PropertySetter<T> setter)
+        {
+            ConfigurationUnit<?> old = registry.get(name);
+            registry.replace(name, new ConfigurationUnit<>(old.source, name, old.getter, clazz, setter));
+        }
+    }
+
+    /**
+     * Configuration unit is a single configuration entry that can be read and written.
+     * @param <V> type of the configuration setter value.
+     */
+    private static class ConfigurationUnit<V> implements Map.Entry<String, V>
```

Comment:
im not seeing why we have this class... it looks like it just duplicates `Property`.